### ISSUE TITLE
Add DDLm checks to GitHub actions

### ISSUE THREAD

**Issue by vaitkus** [COMCIFS/magnetic_dic#30]
Currently, GitHub actions only check the syntax of the dictionary file, but not the style or semantics as in other dictionary repositories.

However, it was mentioned in a comment of a different issue [1] that there are some changes lined up, so I am not completely sure when would be the least disruptive time to turn on the checks. One the one hand they do detected some serious internal inconsistencies, but on the other hand they do complain about some trivial things like update dates, etc.     

[1] https://github.com/COMCIFS/magnetic_dic/issues/29#issuecomment-1610278019

**Comment by brantonc** [comment]
Can GitHub Actions be selectively turned off again for the magCIF repository if it creates a hindrance to pending changes from the commission?

**Comment by vaitkus** [comment]
Yes, in that case we could simply disable the checks. Also, the general policy towards GitHub actions in other dictionary repositories is pretty lax -- even if the checks fail (i.e. issues are detected), this does not technically prevent the PRs from being merged.
